### PR TITLE
[WIP] Polymer 2 support

### LINKED ISSUE #61: Polymer 2?
Has this element (or all the Gold elements for that concern) been deprecated?
If not will they be upgraded to Polymer 2?
I don't see any branch letting me suppose so at the moment

### PR DESCRIPTION
**Pull request by gurpreetatwal**
Taking a stab at upgrading this element to support Polymer 2, taking heavy inspiration from the other elements that have already been upgraded.

fixes #61

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
+    - secure: OJAW+ssXIXb+2Ze4skRm4QrFY6OcpFgKHRhcdT5TGv6OpzHPZO5wWgX37Ocb2FtgGbiIAMiI1B6dR4C998LGOUVO7kGA1f7vnp75PvsvYON1sKLfAyuIBJuI6uLRE6jOSUDAG9Rc3Q3Hv33QOnDcC1uwuYDFIF8pjn++TPqx3c0WwapD+IBsnHhFsTi3H7d0vwJNRUprAZq4leo+3f/jz4PMWlV7IxrjG0ghseOO4sEXDLkm6GXeaJV78HCoNFk/14LC947BxN6T5Sc9lzWo/s0t8JoZODjAe6uON9OzfgsCs8oH7hhNys8QM/6EqBgcL2sEvJi9N0vsBKG351DHVFjGMv7j28+N/FzYkrrtET6EyPxcZ7S3TYUQb92hW/cMBJ6lqusmGfzFQ56WnAJyKJei0vH+IykBFBJ+Km2X3DS3/i30l0j8w4dN0vjYkAC+oWGho5iDN8cp5K93cLhgnDhPr6fU2tRMHB3bPjRMGgT+TmP9zJJ87mSiwCJgJJGl8NVODz54e66CZcOPZdTBzsdC7YSfvlvUE7PUt77LJO69EfZT3s3CAfe0hks7fFpZ8SIqw50F2K5ed7o4CswewPlvsA0yeZCOHZWFk0X2rRSWzfgz7TOAfim37HYp/kYQWFi8XYpd93Q5uPSVId6CX/dJoGfFNHx4oY/lsKQKbYo=
```
Comment (gurpreetatwal) [inline .travis.yml]: these were supposed to change in #60, not sure if they still need to change or not

File: date-input.html [hunk 2]
```diff
@@ -53,40 +53,85 @@
...
+            disabled$="[[disabled]]"
+            invalid="{{invalid}}"
+            inputmode$="[[inputmode]]"
+            readonly$="[[readonly]]">
+      </div>
+    </template>
+
+    <template id="v2">
+
+      <iron-input id="expirationMonth" slot="monthInput"
```
Comment (robrez) [inline date-input.html]: the v2 template is missing the `<div class="container">`... maybe that container should be pulled outside of the version-specific templates and into the main template?

Comment (robrez) [inline date-input.html]: `slot` isn't needed here

File: date-input.html [hunk 3]
```diff
@@ -53,40 +53,85 @@
...
+      </iron-input>
+
       <span>/</span>
-      <input is="iron-input"
-          id="expirationYear"
-          aria-labelledby$="[[_computeAriaLabel(ariaLabelPrefix,'yearLabel')]]"
-          required$="[[required]]"
-          maxlength="2"
+
+      <iron-input id="expirationYear" slot="yearInput"
```
Comment (robrez) [inline date-input.html]: `slot` isn't needed here

File: gold-cc-expiration-input.html [hunk 4]
```diff
@@ -66,7 +66,7 @@
         disabled$="[[disabled]]"
         invalid="[[invalid]]">
 
-      <label hidden$="[[!label]]">[[label]]</label>
+      <label slot="label" hidden$="[[!label]]">[[label]]</label>
 
       <date-input class="paper-input-input" id="input"
```
Comment (robrez) [inline gold-cc-expiration-input.html]: `slot="input"` is needed here

**Comment by robrez** [comment]
I've also just noticed that there is significant a11y problem.

`tab` isn't setting focus on the individual inputs anymore (at least under polymer 2 + native shadowdom). I'm not sure what - if anything you can really do about this without changing the overall design.

**Comment by gurpreetatwal** [comment]
closing PR as it out of date and I no longer user Polymer

**Pull request closed without merging**